### DISCUSSION TITLE
Strange line in CMakeLists.txt

### DISCUSSION THREAD

**Discussion by chetan-set** [IfcOpenShell/IfcOpenShell#3208] [General]
Hello,

I would like to comment on line  [line 535](https://github.com/IfcOpenShell/IfcOpenShell/blob/v0.7.0/cmake/CMakeLists.txt#L535) in the CMakeLists.txt file which refers to the start of this statement:

`        SET(HDF5_LIBRARIES`
`           /usr/lib/x86_64-linux-gnu/hdf5/serial/libhdf5_cpp.so`
`            /usr/lib/x86_64-linux-gnu/hdf5/serial/libhdf5.so`
`            /usr/lib/x86_64-linux-gnu/libsz.so`
`            /usr/lib/x86_64-linux-gnu/libaec.so`
`            z dl`
`        )`

This is the code branch that is reached if we do not find HDF5 with cmake find_library. It turns out that I end up on this code branch. The issue is that I'm on OSX. And it's looking for *.so files. 

I would like to create a pull request that uses the following code snippet (found in the same Cmake file)

```
            if (WIN32)
                # Windows
                set(zlib_post zlib)
                set(lib_ext lib)
            elseif(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
                # macOS
                set(zlib_post libz)
                set(lib_ext dylib)
            else()
                # linux
                set(zlib_post libz)
                set(lib_ext so)
            endif()
```

The second thing I'd like to point out is the last line of the `SET` statement I shared above. The line in question is:

`z dl`

I'm wondering if that's a typo.

Do give me your thoughts.

**Comment by aothms** [comment]
Hi, the HDF5 library initiation is indeed wonky. Part of the reason is that it includes the possibilities of the win/nix build script which uses the HDF5 ctest install package [0] which is a bit unconventional. That is ctest install package is actually what should be referenced by `HDF5_LIBRARY_DIR`. That's why zlib is referenced as `libzlib_debug.lib` for example, which is in no other environment ever called like that.

If you do not use the HDF5 ctest install package you should simply directly specify `-DHDF5_LIBRARIES` directly (as in [1] step 7).

> I would like to create a pull request that uses the following code snippet (found in the same Cmake file)

That is specifically when building with `conda` when we also know where the libraries  are going to be.

> The second thing I'd like to point out is the last line of the SET statement I shared above. The line in question is:
> 
>     z dl

It's not a typo. Those are also libz and libdl which are dependencies. 

Hope that clarifies things a bit. So, in short, improvements are always appreciated. The HDF5 library search routine is quite strange, but it's also because the naming of hdf5 and dependencies is really inconsistent across platforms.

[0] https://github.com/IfcOpenShell/IfcOpenShell/blob/v0.7.0/nix/build-all.py#L415
[1] https://github.com/IfcOpenShell/IfcOpenShell/blob/v0.7.0/src/ifcopenshell-python/docs/ifcopenshell/installation.rst#compiling-on-linux
